### PR TITLE
Darwinia CC1 (Dev)

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

File: node/service/src/chain_spec.rs [hunk 1]
```diff
@@ -69,9 +93,23 @@ pub fn crab_properties() -> Properties {
 	properties
 }
 
+/// Properties for Darwinia.
+pub fn darwinia_properties() -> Properties {
+	let mut properties = Properties::new();
+
+	properties.insert("ss58Format".into(), 42.into());
```
Comment (wuminzhe) [inline node/service/src/chain_spec.rs:100]: 42 for mainnetdev?

Comment (HackFisher) [reply node/service/src/chain_spec.rs:100]: > 42 for mainnetdev?

good catch

File: node/service/src/chain_spec.rs [hunk 2]
```diff
@@ -309,49 +347,171 @@ pub fn crab_build_spec_genesis() -> CrabGenesisConfig {
...
+			validator_count: 15,
+			stakers: vec![(
+				genesis_validator.0.clone(),
+				genesis_validator.1.clone(),
+				COIN,
+				darwinia_runtime::StakerStatus::Validator
+			)],
+			force_era: darwinia_runtime::Forcing::NotForcing,
+			slash_reward_fraction: Perbill::from_percent(10),
+			payout_fraction: Perbill::from_percent(50),
```
Comment (wuminzhe) [inline node/service/src/chain_spec.rs:469]: 50% or 40%?

File: node/service/src/chain_spec.rs [hunk 3]
```diff
@@ -309,49 +347,171 @@ pub fn crab_build_spec_genesis() -> CrabGenesisConfig {
...
+			)]
+		}),
+		pallet_grandpa: Some(Default::default()),
+		pallet_im_online: Some(Default::default()),
+		pallet_authority_discovery: Some(Default::default()),
+		pallet_collective_Instance0: Some(Default::default()),
+		pallet_collective_Instance1: Some(Default::default()),
+		darwinia_elections_phragmen: Some(Default::default()),
+		pallet_membership_Instance0: Some(Default::default()),
+		darwinia_vesting: Some(Default::default()),
```
Comment (wuminzhe) [inline node/service/src/chain_spec.rs:491]: add vestings when not dev?

File: node/service/src/chain_spec.rs [hunk 4]
```diff
@@ -309,49 +347,171 @@ pub fn crab_build_spec_genesis() -> CrabGenesisConfig {
...
+		pallet_sudo: Some(darwinia_runtime::SudoConfig { key: root_key }),
+		darwinia_ethereum_backing: Some(darwinia_runtime::EthereumBackingConfig {
+			// Los Angeles: 15/09/2020, 23:42:08
+			// Berlin :     16/09/2020, 14:42:08
+			// Beijing:     16/09/2020, 13:42:08
+			// New York :   16/09/2020, 01:42:08
+			ring_locked: 1_179_562_684_772_882_521,
+			kton_locked: 55_784_589_946_137,
+			..Default::default()
```
Comment (wuminzhe) [inline node/service/src/chain_spec.rs:500]: token address?

File: node/service/src/chain_spec.rs [hunk 5]
```diff
@@ -309,49 +347,171 @@ pub fn crab_build_spec_genesis() -> CrabGenesisConfig {
...
+		darwinia_elections_phragmen: Some(Default::default()),
+		pallet_membership_Instance0: Some(Default::default()),
+		darwinia_vesting: Some(Default::default()),
+		pallet_sudo: Some(darwinia_runtime::SudoConfig { key: root_key }),
+		darwinia_ethereum_backing: Some(darwinia_runtime::EthereumBackingConfig {
+			// Los Angeles: 15/09/2020, 23:42:08
+			// Berlin :     16/09/2020, 14:42:08
+			// Beijing:     16/09/2020, 13:42:08
+			// New York :   16/09/2020, 01:42:08
+			ring_locked: 1_179_562_684_772_882_521,
```
Comment (wuminzhe) [inline node/service/src/chain_spec.rs:498]: This value is not calculated based on mapped rings?

File: runtime/darwinia/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,1132 @@
...
+				// for tips, if any, 80% to treasury, 20% to author (though this can be anything)
+				tips.ration_merge_into(80, 20, &mut split);
+			}
+			Treasury::on_unbalanced(split.0);
+			ToAuthor::on_unbalanced(split.1);
+		}
+	}
+}
+parameter_types! {
+	pub const TransactionByteFee: Balance = 500 * MICRO;
```
Comment (wuminzhe) [inline runtime/darwinia/src/lib.rs:272]: 0.0005 RING

**Comment by AurevoirXavier** [comment]
@HackFisher Please provide 2 vesting account.

**Comment by HackFisher** [comment]
> @HackFisher Please provide 2 vesting accounts.

Sent to you through PM

**Comment by HackFisher** [comment]
> @HackFisher Please provide 2 vesting accounts.

Sent to you through PM

**Review by HackFisher — CHANGES_REQUESTED** [review]
Better to bump crab runtime version too, cause there are changes in common.

**Review by HackFisher — APPROVED** [review]

**Pull request merged**